wallet-core/src/lib.rs

```diff
@@ -10,57 +10,36 @@
 #![deny(missing_docs)]
 #![deny(rustdoc::broken_intra_doc_links)]
 #![deny(clippy::pedantic)]
+#![feature(try_trait_v2)]
+
 #[cfg(target_family = "wasm")]
 #[global_allocator]
 static ALLOC: dlmalloc::GlobalDlmalloc = dlmalloc::GlobalDlmalloc;
 
 extern crate alloc;
 
-pub mod keys;
-pub mod transaction;
-
 #[cfg(target_family = "wasm")]
+#[macro_use]
 mod ffi;
 
-/// Length of the seed of the generated rng.
-pub const RNG_SEED: usize = 64;
+pub mod keys;
+pub mod notes;
+pub mod transaction;
+
+/// The seed used to generate the entropy for the keys
+pub type Seed = [u8; 64];
 
 // The maximum amount of input notes that can be spend in one
 // phoenix-transaction
 const MAX_INPUT_NOTES: usize = 4;
 
-use alloc::collections::btree_map::BTreeMap;
 use alloc::vec::Vec;
 
 use dusk_bytes::{DeserializableSlice, Serializable, Write};
 
-use execution_core::{
-    transfer::phoenix::{
-        Note, NoteLeaf, SecretKey as PhoenixSecretKey,
-        ViewKey as PhoenixViewKey,
-    },
-    BlsScalar,
-};
-
-/// Filter all notes and their block height that are owned by the given keys,
-/// mapped to their nullifiers.
-pub fn map_owned(
-    keys: impl AsRef<[PhoenixSecretKey]>,
-    notes: impl AsRef<[NoteLeaf]>,
-) -> BTreeMap<BlsScalar, NoteLeaf> {
-    notes
-        .as_ref()
-        .iter()
-        .fold(BTreeMap::new(), |mut notes_map, note_leaf| {
-            for sk in keys.as_ref() {
-                if sk.owns(note_leaf.note.stealth_address()) {
-                    let nullifier = note_leaf.note.gen_nullifier(sk);
-                    notes_map.insert(nullifier, note_leaf.clone());
-                }
-            }
-            notes_map
-        })
-}
+use execution_core::transfer::phoenix::{Note, ViewKey as PhoenixViewKey};
+
+pub use notes::map_owned;
 
 /// Calculate the sum for all the given [`Note`]s that belong to the given
 /// [`PhoenixViewKey`].
```

Comment:
Imo the function `phoenix_balance` can also be moved to the `notes` module since it calculates the balance of notes.